client/src/schema/selection.ts

```diff
@@ -74,20 +74,24 @@ type NestedValueAtColumn<O, Key extends SelectableColumn<O>> =
     ? N extends DataProps<O>
       ? {
           [K in N]: M extends SelectableColumn<NonNullable<O[K]>>
-            ? NestedValueAtColumn<NonNullable<O[K]>, M> & XataRecord
+            ? NonNullable<O[K]> extends XataRecord
```

Comment:
This new check adds ``XataRecord`` only for items that are records (not object types). Re #157